cmd/transflect-operator/operator.go

```diff
@@ -96,20 +81,36 @@ func watchErrorHandler(_ *cache.Reflector, err error) {
 }
 
 func (o *operator) start() error {
+	// Initialise activeState for existing transflect EnvoyFilters
+	// to determine if EnvoyFilter upsert should be processed.
 	if err := o.syncActive(); err != nil {
 		return fmt.Errorf("cannot start operator: %w", err)
 	}
 
+	// Initialise work-queue and stopper channel
+	o.queue = workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+	o.stopper = make(chan struct{})
+
+	// Initialise informers
+	informerFactory := informers.NewSharedInformerFactory(o.k8s, time.Second*30)
+	o.rsInformer = informerFactory.Apps().V1().ReplicaSets()
+	if err := o.rsInformer.Informer().SetWatchErrorHandler(watchErrorHandler); err != nil {
+		return errors.Wrap(err, "cannot add custom error handler to replicaset informer")
+	}
 	o.rsInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc:    o.enqueue,
 		UpdateFunc: o.enqueueWithOld,
 	})
-
+	o.deployInformer = informerFactory.Apps().V1().Deployments()
+	if err := o.deployInformer.Informer().SetWatchErrorHandler(watchErrorHandler); err != nil {
+		return errors.Wrap(err, "cannot add custom error handler to deployment informer")
+	}
 	o.deployInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
 		DeleteFunc: o.cleanupDeployment,
 	})
+	log.Debug().Msg("Start leading: informer event handlers registered")
 
-	log.Debug().Msg("Starting: informer event handlers registered")
+	// Run workers and informers
 	o.runWorkers(42)
 	o.rsInformer.Informer().Run(o.stopper)
 	o.deployInformer.Informer().Run(o.stopper)
```

Comment:
These two calls to `Run()` are not done asynchronously. From the Run docs:

> Run starts and runs the shared informer, returning after it stops.

So the deployInformer does not start until the rsInformer stops.

Reply:
yesss!! thanks for the eagle eye yet again.